File: admins/src/main/java/com/github/admins/services/impl/CategoryService.java
package com.github.admins.services.impl;

import com.github.admins.dto.CategoryDto;
import com.github.admins.services.ICategoryService;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class CategoryService implements ICategoryService {

    @Override
    public Optional<CategoryDto> create(CategoryDto c) {
        return Optional.empty();
    }

    @Override
    public Optional<CategoryDto> readByName(String categoryName) {
        return Optional.empty();
    }

    @Override
    public Optional<List<CategoryDto>> readAll() {
        return Optional.empty();
    }

    @Override
    public void update(CategoryDto c) {

    }

    @Override
    public void delete(Long id) {

    }
}

File: server-mocks/src/main/kotlin/com/github/server/mocks/services/impl/CategoryService.kt
package com.github.server.mocks.services.impl

import com.github.server.mocks.payload.Category
import com.github.server.mocks.services.ICategoryService
import com.github.server.mocks.utils.IdsGenerator
import org.springframework.stereotype.Service

@Service
class CategoryService(
        private val idsGenerator: IdsGenerator
): ICategoryService {

    override fun create(category: Category): Category {
        category.id = idsGenerator.getId()
        return category
    }

    override fun readByName(name: String): Category {
        return Category(idsGenerator.getId(), "Phone")
    }

    override fun readAll(): List<Category> {
        return listOf(
                Category(idsGenerator.getId(), "Phone"),
                Category(idsGenerator.getId(), "Computer"),
                Category(idsGenerator.getId(), "Soft"),
                Category(idsGenerator.getId(), "Instruments"),
                Category(idsGenerator.getId(), "Sport")
        )
    }

    override fun update(category: Category) {}

    override fun remove(id: Long) {}


}